app/controllers/api/v1/targets_controller.rb

```diff
@@ -12,6 +12,10 @@ def show; end
 
       def create
         @target = current_user.targets.create!(target_params)
+        @targets = Target.match_targets(@target)
+        devices = @targets.flat_map { |target| target.user.devices }.uniq
```

Comment:
If you need just the ids I think you can do something like `@devices_ids = @targets.flat_map { |target| target.user.devices.pluck(:id) }.uniq`